### ISSUE TITLE
Make the JNI lib name configurable?

### ISSUE THREAD

**Issue by nirvdrum** [jni-rs/jni-rs#417]
The GraalVM project from Oracle Labs ships with a tool called Native Image that can ahead-of-time compile Java applications to native executables or libraries. The resulting binary also provides an implementation of JNI, so you could build a native library of your Java code and interact with it via the JNI Invocation API. I experimented with using this mechanism to embed Truffle languages in native libraries suitable for embedding in native applications ([write-up here](https://nirvdrum.com/2022/05/09/truffle-language-embedding.html) if interested).

Currently, the project uses _java-locator_ to do that lookup and that only looks for a file named "libjvm" or thereabouts. I'd like to be able to set an environment variable or use an API call to specify the name or path to an alternative library that is JNI-compatible. I think it makes sense here instead of _java-locator_ because that project is really about interacting with Java installations, which a native image is not.

I'd be happy to take a crack at implementing this, but I wanted to see if there'd be buy-in before beginning the work and, if so, if there are any design considerations or preferences in approach.

**Comment by argv-minus-one** [comment]
You can use [`JavaVM::with_libjvm`](https://github.com/jni-rs/jni-rs/blob/a0c4c48f7b4304a4b109d3617213159c142d64c1/src/wrapper/java_vm/vm.rs#L172) to load an arbitrary shared library as the JVM. Does that help?

**Comment by rib** [comment]
Hi @nirvdrum, if you get a moment can you maybe comment on whether  [`JavaVM::with_libjvm`](https://github.com/jni-rs/jni-rs/blob/a0c4c48f7b4304a4b109d3617213159c142d64c1/src/wrapper/java_vm/vm.rs#L172) is enough for your use case?

**Comment by nirvdrum** [comment]
I'm sorry, I got swamped at work. I suspect this is what I'm looking for, but I'm trying to build a new native image with GraalVM 22.3.1 to verify. I should have that wrapped up in a day or two.

**Comment by nirvdrum** [comment]
Thanks for being patient. `with_libjvm` does indeed work. I went back through to see how I missed it. I think it's because I started with 0.20, which didn't have that method, and then didn't see `with_libjvm` was added with 0.21 (still in dev when I opened this). Be that as it may, the new `with_libjvm` solves my problem and I'm happy to see it plays nicely with Native Image compiled libraries.

Separate from this, I'm going to pull together a PR with some example usages of the Invocation API.
